man/journal-upload.conf.xml

```diff
@@ -60,6 +60,31 @@
         <xi:include href="version-info.xml" xpointer="v232"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>Compression=</varname></term>
+
+        <listitem><para><literal>:</literal>-separated values of compression algorithm and level to be applied to logs data before sending.
+        Key one of <literal>none</literal>, <literal>zstd</literal>, <literal>xz</literal>, and <literal>lz4</literal>.
+        Defaults to <literal>none</literal>. Value is expected to be a positive integer.
+        Defaults to unset, and the default compression level of the specified algorithm will be used.</para>
+
+        <para>Even with compression enabled initial requests are sent without compression. It becomes effective either if
+        <literal>ForceCompression=</literal> is set to <literal>true</literal> or server response contains
+        <literal>Accept-Encoding</literal> headers with list of compression algorithms, which are configured and supported by
+        systemd-journal-upload.</para>
+
+        <xi:include href="version-info.xml" xpointer="v258"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>ForceCompression=</varname></term>
```

Comment:
I do not think we should introduce such config. Always negotiate, and compress only when the receiver accepts compression.

Or, could you elaborate the use case?

Reply:
Added this options for scenario when no negotiation is needed or supported by server, but server supports configured on systemd compression algorithms

Reply:
When you know both sender and receiver supports compression, then negotiation may not be necessary, but at the same time, not necessary to skip negotiation.
Or, negotiation has some problem? Performance issue? Why you want to skip it?

Reply:
when during negotiation no compression algorithms are returned no compression will be applied. also it requires to add this implementation to all log storages, which support journal logs. instead we can just send compressed data immediately expecting server to support this

Reply:
But it may be simply refused. And, IIUC, when it is refused, journal-upload will not try with another compression algorithm or without compression.